server/src/test/kotlin/org/jetbrains/bsp/bazel/server/sync/languages/rust/RustDependencyResolverTest.kt

```diff
@@ -0,0 +1,140 @@
+package org.jetbrains.bsp.bazel.server.sync.languages.rust
+
+import ch.epfl.scala.bsp4j.RustPackage
+import io.kotest.matchers.collections.shouldContainExactlyInAnyOrder
+import io.kotest.matchers.maps.shouldContainExactly
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+import org.jetbrains.bsp.bazel.bazelrunner.BasicBazelInfo
+import org.jetbrains.bsp.bazel.bazelrunner.BazelRelease
+import org.jetbrains.bsp.bazel.server.sync.BazelPathsResolver
+import org.jetbrains.bsp.bazel.server.sync.model.*
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.net.URI
+import java.nio.file.Paths
+
+class RustDependencyResolverTest {
+
+    private val outputBase = "/private/var/tmp/_bazel/125c7a6ca879ed16a4b4b1a74bc5f27b"
+    private val execRoot = "$outputBase/execroot/bazel_bsp"
+    private lateinit var rustPackageResolver: RustPackageResolver
+    private lateinit var resolver: RustDependencyResolver
+
+    @BeforeEach
+    fun beforeEach() {
+        // given
+        val bazelInfo = BasicBazelInfo(
+            execRoot = execRoot,
+            outputBase = Paths.get(outputBase),
+            workspaceRoot = Paths.get("/Users/user/workspace/bazel-bsp"),
+            release = BazelRelease.fromReleaseString("release 6.0.0")
+        )
+
+        rustPackageResolver = RustPackageResolver(BazelPathsResolver(bazelInfo))
+        resolver = RustDependencyResolver(rustPackageResolver)
+    }
+
+    @Test
+    fun `should return empty dependency list for empty package list`() {
+        // given
+        val modules = emptyList<Module>()
+        val packages = emptyList<RustPackage>()
+
+        // when
+        val (dependencies, rawDependencies) = resolver.rustDependencies(packages, modules)
+
+        // then
+        dependencies shouldNotBe null
+        dependencies shouldBe emptyMap()
```

Comment:
the first check is not needed there i think

Reply:
Yep, I deleted it